src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Options/OptionsStorage.cs

```diff
@@ -13,18 +13,22 @@
 using Microsoft.Internal.VisualStudio.Shell.Interop;
 using Microsoft.VisualStudio.Utilities.UnifiedSettings;
 using System.Linq;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Shell.Interop;
+using System.Threading.Tasks;
 
 namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Options;
 
-[Shared]
+[System.Composition.Shared]
```

Comment:
Is this actually needed? I thought `System.ComponentModel.Composition` exports were shared by default and MEFv2 (`System.Composition`) require `Shared`.

Reply:
oh probably not, I always forget they're shared by default in V2

Reply:
Actually, the problem is that they're _not_ shared by default in V2. V2 is `System.Composition`. So confusing... :smile:

Reply:
WTF! 🤯🤯🤯

I had _no_ idea that components were shared by default in _any_ MEF version.

Reply:
Oh yea, the fact that versions just use a different namespace is a headache and a half

Reply:
I literally just thought it was a "we need new attributes so we need a new namespace" bit of legacy, I didn't think there was a behaviour difference. Ugh. I've definitely written the fully qualified Shared attribute thing a few times, thats for sure

Reply:
Yup. It's true. Older MEF was shared by default. When @jasonmalinowski switched the lower layers of Roslyn over to MEFv2 (i.e. "portable MEF"), we had to get used to adding `[Shared]` everywhere in those layers. In Roslyn, there generally aren't any `[Shared]` attributes in the Editor Features layer and up because it's using good 'ol MEF v1 (i.e. .NET Framework MEF, System.ComponentModel.Composition, etc.). You can see this in lots of places, such as [this one](https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/CSharp/EncapsulateField/EncapsulateFieldCommandHandler.cs).

VS MEF does the work to support both sets of attributes.